crates/yttrium/src/pay/mod.rs

```diff
@@ -22,72 +27,894 @@ impl<T: std::fmt::Debug> From<progenitor_client::Error<T>> for PayError {
+    ) -> Result<Vec<RequiredAction>, GetPaymentRequestError> {
+        let cache = self.cached_options.read().map_err(|e| {
+            GetPaymentRequestError::InternalError(format!(
+                "Cache lock poisoned: {}",
+                e
+            ))
+        })?;
+
+        let cached_option = cache
+            .iter()
+            .find(|o| o.option_id == option_id)
+            .ok_or_else(|| {
+                GetPaymentRequestError::OptionNotFound(format!(
+                    "Option {} not found in cache. Call get_payment_options first.",
+                    option_id
+                ))
+            })?;
+
+        Ok(cached_option.required_actions.clone())
+    }
+
+    /// Fetch an action for a payment option
+    pub async fn fetch(
+        &self,
+        payment_id: String,
+        option_id: String,
+        data: String,
+    ) -> Result<FetchResponse, PayError> {
+        let data_value: serde_json::Value =
+            serde_json::from_str(&data).unwrap_or(serde_json::Value::Null);
+        let body = types::FetchRequest { option_id, data: data_value };
+        let response = with_retry(|| async {
+            self.client
+                .fetch_handler()
+                .api_key(&self.config.api_key)
+                .sdk_name(&self.config.sdk_name)
+                .sdk_version(&self.config.sdk_version)
+                .sdk_platform(&self.config.sdk_platform)
+                .id(&payment_id)
+                .body(body.clone())
+                .send()
+                .await
+        })
+        .await?;
+        Ok(response.into_inner().into())
+    }
+
+    /// Get gateway payment status (for polling)
+    pub async fn get_gateway_payment_status(
+        &self,
+        payment_id: String,
+        max_poll_ms: Option<i64>,
+    ) -> Result<GetPaymentStatusResponse, PayError> {
+        let mut req = self
+            .client
+            .gateway_get_payment_status()
+            .api_key(&self.config.api_key)
+            .sdk_name(&self.config.sdk_name)
+            .sdk_version(&self.config.sdk_version)
+            .sdk_platform(&self.config.sdk_platform)
+            .id(&payment_id);
+        if let Some(ms) = max_poll_ms {
+            req = req.max_poll_ms(ms);
+        }
+        let response =
+            with_retry(|| async { req.clone().send().await }).await?;
+        Ok(response.into_inner().into())
+    }
+
+    /// Confirm a payment
+    pub async fn confirm_payment(
+        &self,
+        payment_id: String,
+        max_poll_ms: Option<i64>,
+    ) -> Result<ConfirmPaymentResponse, ConfirmPaymentError> {
+        let body = types::ConfirmPaymentRequest(serde_json::Map::new());
+        let mut req = self
+            .client
+            .confirm_payment_handler()
+            .api_key(&self.config.api_key)
+            .sdk_name(&self.config.sdk_name)
+            .sdk_version(&self.config.sdk_version)
+            .sdk_platform(&self.config.sdk_platform)
```

Comment:
those params are the same on every api call, can we somehow preconfigure the client to always include them?